### PR TITLE
ISPN-5039 Add failover tests simulating node crash/kill

### PR DESCRIPTION
**Pull request by vblagoje**
Master only. Additional failover tests specifically designed to test node kill and failover.

### CODE REVIEW

File: core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      private long timeout;
+
+      public SleepingSimpleCallable(long timeout) {
+         super();
+         this.timeout = timeout;
+      }
+
+      @Override
+      public Integer call() throws Exception {
+         Thread.sleep(timeout);
```
Comment (galderz) [inline core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java]: Right, but sleeps like this are sometime the cause why tests fail randomly. Plus, what is the exact assumption of this test in terms of events? I guess `killCacheManagers` should be called while the callable is sleeping? What would happen if GC hit the testsuite and the callable finished (sleep finished) before `killCacheManagers` was called?

Dan's suggestion helps answers these questions and should make the test more predictable.

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Pull requests monitor [Build 8187/vblagoje:t_5039](http://ci.infinispan.org/viewLog.html?buildId=22547&buildTypeId=bt9&guest=1) for bbd9472c287ef5d2c2fb535b11a422b3ba87289a
Tests: 11216, 1 failed (1 new), 23 ignored. Build time: 02:08:09

Failed tests detected
* (new) TestSuite: org.infinispan.persistence.sifs.SoftIndexFileStoreTest.testLoadAndStoreImmortal



Error message is logged

Build duration (secs) is too large: 7672

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Pull requests monitor [Build 8188/vblagoje:t_5039](http://ci.infinispan.org/viewLog.html?buildId=22550&buildTypeId=bt9&guest=1) for bbd9472c287ef5d2c2fb535b11a422b3ba87289a
Tests: 11216, 2 failed (2 new), 23 ignored. Build time: 01:51:39

Failed tests detected
* (new) TestSuite: org.infinispan.partitionhandling.EntryRetrieverReplPartitionHandlingTest.testRetrievalWhenPartitionIsDegrated


* (new) org.infinispan.server.test.query.ManualIndexingIT.testManualIndexing



Error message is logged

Build duration (secs) is too large: 6669

File: core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java (hunk 1 above)
Comment (vblagoje) [inline core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java]: @galderz and @danberindei I followed your advice. Have a look please. Thank you both

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Pull requests monitor [Build 8205/vblagoje:t_5039](http://ci.infinispan.org/viewLog.html?buildId=22568&buildTypeId=bt9&guest=1) for ea9e1c287824f7eff48d437d7cffd3a5590d47f0
Tests: 11217, 0 failed (0 new), 23 ignored. Build time: 02:11:51

Error message is logged

Build duration (secs) is too large: 7894

**Comment by danberindei** [comment]
Integrated, thanks Vladimir!

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Pull requests monitor [Build 8228/vblagoje:t_5039](http://ci.infinispan.org/viewLog.html?buildId=22618&buildTypeId=bt9&guest=1) for ea9e1c287824f7eff48d437d7cffd3a5590d47f0
Tests: 11217, 5 failed (5 new), 23 ignored. Build time: 01:41:40

Failed tests detected
* (new) TestSuite: org.infinispan.statetransfer.ClusterTopologyManagerTest.testClusterRecoveryAfterThreeWaySplit


* (new) TestSuite: org.infinispan.interceptors.ConcurrentInterceptorVisibilityTest.testSizeVisibility


* (new) TestSuite: org.infinispan.client.hotrod.query.MultiHotRodServerIspnDirReplQueryTest.testAttributeQuery


* (new) TestSuite: org.infinispan.client.hotrod.query.MultiHotRodServerIspnDirReplQueryTest.testEmbeddedAttributeQuery


* (new) TestSuite: org.infinispan.client.hotrod.query.MultiHotRodServerIspnDirReplQueryTest.testProjections

**Pull request closed without merging**